controllers/hostingautoscalingpolicy/hostingautoscalingpolicy_controller.go

```diff
@@ -141,7 +142,7 @@ func (r *Reconciler) reconcileHostingAutoscalingPolicy(ctx reconcileRequestConte
 	}
 
 	if err = r.initializeContext(&ctx); err != nil {
-		return r.updateStatusAndReturnError(ctx, FailedAutoscalingJobStatus, errors.Wrap(err, "Unable to initialize operator"))
+		return r.updateStatusAndReturnError(ctx, errors.Wrap(err, "Unable to initialize operator"))
```

Comment:
Why is this calling `updateStatusAndReturnError` if it's not updating the status?

Reply:
same question

Reply:
Check the method. I have added logic there to decide what status to update to.

Reply:
I could still pass this parameter and change status to reconciling if you prefer but since all the calls were with the same status param, this made sense.

Reply:
got it